expressServer/scripts/generateUsernames.js

```diff
@@ -30,20 +21,18 @@ const generateRandomUsers = async (numUsers, batchSize) => {
 
     if (users.length === batchSize) {
       await User.insertMany(users);
-      console.log(`Batch of ${batchSize} users saved to the database`);
       users = [];
     }
   }
 
   if (users.length > 0) {
     await User.insertMany(users);
-    console.log(`Final batch of ${users.length} users saved to the database`);
   }
 };
 
-generateRandomUsers(10000, 10)
+generateRandomUsers(1000, 200)
   .then(() => {
     console.log('Random users generated and saved to the database');
   })
   .catch(err => console.error('Error generating users:', err))
-  .finally(() => mongoose.connection.close());
+  .finally(() => closeDB());
```

Comment:
`closeDB()` is async. Connection coult not be closed. 

Also you are missing the `process.exit(0)` to set the script finished.